blob/service.go

```diff
@@ -24,20 +28,28 @@ var (
 	log = logging.Logger("blob")
 )
 
+const maxRetries = 5
+
 // Submitter is an interface that allows submitting blobs to the celestia-core. It is used to
 // avoid a circular dependency between the blob and the state package, since the state package needs
 // the blob.Blob type for this signature.
 type Submitter interface {
-	SubmitPayForBlob(ctx context.Context, fee math.Int, gasLim uint64, blobs []*Blob) (*types.TxResponse, error)
+	SubmitPayForBlob(ctx context.Context, fee sdkmath.Int, gasLim uint64, blobs []*Blob) (*types.TxResponse, error)
+	QueryMinimumGasPrice(context.Context) (float64, error)
 }
 
 type Service struct {
 	// accessor dials the given celestia-core endpoint to submit blobs.
 	blobSumitter Submitter
 	// shareGetter retrieves the EDS to fetch all shares from the requested header.
 	shareGetter share.Getter
-	//  headerGetter fetches header by the provided height
+	// headerGetter fetches header by the provided height
 	headerGetter func(context.Context, uint64) (*header.ExtendedHeader, error)
+	// minGasPrice is the minimum gas price that the node will accept.
+	// NOTE: just because the first node accepts the transaction, does not mean it
+	// will find a proposer that does accept the transaction. Better would be
+	// to set a global min gas price that correct processes conform to.
+	minGasPrice float64
```

Comment:
I believe this should be part of the state accessor so that other services and transactions can rely on gas price tweaks introduced here. This would also remove the need for the QueryMinimuGasPrice interface method usage of which is better be implementation detail of general tx submission code path.

Reply:
yeah I thought about that initially. I wasn't sure if `core.Accessor` was just supposed to be some "dumb" quasi stateless access point to the consensus node or whether it was supposed to have logic within it. I guess if it's already doing nonce counting then perhaps it's okay. 

I can shift it across but we should be careful about overloading a single component with too much functionality.